### PR TITLE
PWGHF: Fix for Signal status in MC Rec in D0-hadron correlations and implement event mixing

### PR DESCRIPTION
**Pull request by samrangy**

### CODE REVIEW

**Review by vkucera — CHANGES_REQUESTED** [review]
Hi @samrangy , it seems you want to use `signalStatus` as a bitmap. In that case, it is better to define your bits as named `enum` values and then use the macros `SETBIT` and `TESTBIT` for the bitwise operations with `signalStatus`.

**Comment by gluparel** [comment]
Hello @vkucera, all,

can we try to define a timescale for this pull request? Samuele was actually waiting for the code of shyam since some time (shyam original pull request on EM dates back to more that one month ago) and at the end we decided to go ahead with this pull request to speed up the things.

Samuele is now stuck waiting for the code since some time. Since he is a master student he doesn't have an infinite amount of time for developing the code and a clear timescale would help him. 

Many thanks for your feedback!
Grazia

**Comment by vkucera** [comment]
Converted to draft because of inactivity.

**Comment by fcolamar** [comment]
Hi @samrangy, any update on this PR?

**Comment by vkucera** [comment]
@mmazzilli @deepathoms @fcolamar Can you please review this?

**Comment by vkucera** [comment]
Conflicts need to be resolved.

File: PWGHF/HFC/TableProducer/correlatorD0Hadrons.cxx [hunk 1]
```diff
@@ -402,47 +498,49 @@ struct HfCorrelatorD0Hadrons {
 
         int signalStatus = 0;
         if ((flagD0 == true) && (candidate1.isSelD0() >= selectionFlagD0) && (isSoftpiD0 == false)) {
-          signalStatus += 1;
+          SETBIT(signalStatus, o2::aod::hf_correlation_d0_hadron::kD0Sig);
```
Comment (vkucera) [inline PWGHF/HFC/TableProducer/correlatorD0Hadrons.cxx:501]:
```suggestion
          SETBIT(signalStatus, ParticleType::kD0Sig);
```
and other occurrences

**Review by vkucera — CHANGES_REQUESTED** [review]

**Comment by vkucera** [comment]
@mmazzilli @deepathoms This PR has been pending since 3 weeks ago.

**Review by mmazzilli — APPROVED** [review]

**Comment by mmazzilli** [comment]
Hello @vkucera, 
@samrangy  informed us that she has modified the naming and the code structure according to the guidelines. Could you please re-open this PR?

Thanks in advance,
Marianna

**Comment by vkucera** [comment]
> Hello @vkucera, @samrangy informed us that she has modified the naming and the code structure according to the guidelines. Could you please re-open this PR so that she can push the new version?
> 
> Thanks in advance, Marianna

Hi @samrangy , The reopen button is blocked because the history of the PR and the history of the [`samrangy:master`](https://github.com/samrangy/O2Physics/tree/master) branch diverged while the PR was closed. @samrangy force-pushed a new commit to her branch which removed the 3 other commits that had been already included in the PR. Force-pushing to branches that have an existing PR is explicitly discouraged in the instructions and there is not simple solution in this case, so I suggest creating a new PR.

**Comment by vkucera** [comment]
Test

**Pull request closed without merging**